src/query/sql/src/planner/binder/ddl/table.rs

```diff
@@ -340,6 +332,32 @@ impl<'a> Binder {
         self.bind_statement(bind_context, &stmt).await
     }
 
+    async fn check_database_exist(&mut self, database: &Option<Identifier<'_>>) -> Result<String> {
+        match database {
+            None => Ok(self.ctx.get_current_database()),
+            Some(ident) => {
+                let database = normalize_identifier(ident, &self.name_resolution_ctx).name;
+                let dbs = self
+                    .ctx
+                    .get_catalog(&self.ctx.get_current_catalog())?
+                    .list_databases(&self.ctx.get_tenant())
```

Comment:
Don't we have a `get_database` API? List all databases to check if it exists seems a bad pattern.